File: clients/ruby/model/vec2_double.rb
class Vec2Double
    attr_accessor :x
    attr_accessor :y
    def initialize(x, y)
        @x = x
        @y = y
    end
    def self.read_from(stream)
        x = stream.read_double()
        y = stream.read_double()
        Vec2Double.new(x, y)
    end
    def write_to(stream)
        stream.write_double(@x)
        stream.write_double(@y)
    end
end

File: clients/python/model/vec2_double.py
class Vec2Double:
    def __init__(self, x, y):
        self.x = x
        self.y = y
    @staticmethod
    def read_from(stream):
        x = stream.read_double()
        y = stream.read_double()
        return Vec2Double(x, y)
    def write_to(self, stream):
        stream.write_double(self.x)
        stream.write_double(self.y)
    def __repr__(self):
        return "Vec2Double(" + \
            repr(self.x) + "," + \
            repr(self.y) + \
            ")"
